### PR TITLE
Generated column doc

### PR DESCRIPTION
**Pull request by Tishj**
This PR adds documentation for generated columns to the duckdb docs.

Syntax for it is added to the Syntax railroad diagram of `create_table`, this portion is made expandable under the name `generated-column`.  

This is what it looks like before it's expanded:
image
And this is after:
image

I've also added a brief description of generated columns to `create_table`. We might want to move this description to its own page later, but I felt this was the best place for it, since generated columns are only supported in `CREATE TABLE` statement for now.

PS:
I've had some issues with a bug where the VIRTUAL|GENERATED options were clipping through the options below, which almost magically disappeared, so please verify that this also works for you by running the site locally with jekyll

### CODE REVIEW

**Comment by Alex-Monahan** [comment]
Looks good! I reordered some sentences a bit just for clarity and I added a few examples. Would you mind double checking my examples? I haven't pulled down the code to test since it hasn't reached master just yet.

**Comment by Tishj** [comment]
Nice! Yes that works, you could also leave off the `GENERATED ALWAYS` part in the example :)

**Comment by Tishj** [comment]
I am not sure about the sentence: "Since an expression has a return type, its type will be automatically derived from the expression." This makes it feel like that will always happen regardless if you specify the type or not, which isn't the case.

**Comment by Alex-Monahan** [comment]
> I am not sure about the sentence: "Since an expression has a return type, its type will be automatically derived from the expression." This makes it feel like that will always happen regardless if you specify the type or not, which isn't the case.

Ah, interesting! Is there a way to know whether or not this will succeed? Or should we just reword it to be more of a "DuckDB will attempt to automatically derive the type from the return type of the expression"?

**Comment by Alex-Monahan** [comment]
Let me know if you think this is ready to merge!

**Comment by Alex-Monahan** [comment]
> I missed your reaction earlier, it should always succeed.
> 
> Actually setting the type explicitly is how you could cause problems, your column might become unusable because the data you insert might not be able to be cast to the type of the generated column that references the column.
> 
> For example, a problematic sequence of actions:
> 
> ```sql
> D CREATE TABLE tbl (
> > col INTEGER,
> > gcol DATE as (col)
> > );
> D INSERT INTO tbl VALUES (5);
> Error: Constraint Error: Constraint Error: Incorrect DATE value for generated column "gcol"
> ```

Ah, now I understand. I will make some adjustments!

**Comment by Alex-Monahan** [comment]
Looks good! Only tweak would be that varient should be variant!
